### PR TITLE
Add a new "char_list" QTreeWidget that allows for categorizing characters

### PR DESCRIPTION
**Pull request by in1tiate**
image
it works

### CODE REVIEW

**Comment by oldmud0** [comment]
I'll hold off on this one until we can come up with a theme or something we can show it off on

**Comment by Crystalwarrior** [comment]
> I'll hold off on this one until we can come up with a theme or something we can show it off on

Do it

**Comment by in1tiate** [comment]
here it is in vanilla. it has occurred to me that it might be prudent to mark which characters are taken
image

**Comment by in1tiate** [comment]
@oldmud0 see https://github.com/AttorneyOnline/AO2-Themes/pull/5

File: src/charselect.cpp [hunk 1]
```diff
@@ -247,24 +299,35 @@ void Courtroom::filter_character_list()
...
     // for the buttons that actually appear.
     // You'd also update the passwordedness and etc. here later.
     current_char->reset();
+    current_char_list_item->setHidden(false);
     current_char->set_taken(char_list.at(i).taken);
+    QString last_four = current_char_list_item->text(0).right(4);
+    if (last_four == " [x]") // reset us
+      current_char_list_item->setText(0, char_list.at(i).name);
+    if (char_list.at(i).taken) // woops, we are taken
+      current_char_list_item->setText(0, current_char_list_item->text(0) + " [x]");
```
Comment (oldmud0) [inline src/charselect.cpp:330]: Why can't you use [`setDisabled()`](https://doc.qt.io/qt-5/qtreewidgetitem.html#setDisabled)? This is the same nasty trick that the pairing menu uses, and it's not a great one.

**Review by oldmud0 — CHANGES_REQUESTED** [review]
Looks great, but just one more thing...

File: src/charselect.cpp (hunk 1 above)
Comment (Crystalwarrior) [reply src/charselect.cpp:330]: I kind of agree, I'd rather setDisabled()

**Comment by stonedDiscord** [comment]
charlist

**Review by oldmud0 — APPROVED** [review]

**Pull request merged**